### DISCUSSION TITLE
completely air-gapped istio deployment on RKE2

### DISCUSSION THREAD

**Discussion by busyboy77** [rancher/rke2#6726] [Q&A] [answered]
My Question is what is the best way to deploy RKE2 helm charts when the RKE2 is deployed in a completely air-gapped environment.

Reasons:

1. We can not have Rancher Management Server deployed 
2. Direct internet access is not possible
3. How to clone RKE2 Istio + Monitoring helm charts ( along with their Images )  on an internet enabled system and then transfer them to air-gapped environments.

We are using RKE2 for private sector deployments because of its compactness and compatibility and ease to use. The RKE2 helm charts are only possible when the internet is available, however, we can't use these RKE2 based helm charts when the deployment is not connected to the direct internet and customer's don't want to allow the internet as well.

For normal helm charts (bitnami etc ) we can clone the OCI based charts and ship them to the air-gapped envs, but the mechanism in RKE2 is so complex ( or I am not aware of anything about them either using HelmChart configs ) how to transport them to offline systems.

**Comment by brandond** [comment, marked as answer]
> We can not have Rancher Management Server deployed
> How to clone RKE2 Istio + Monitoring helm charts ( along with their Images ) on an internet enabled system and then transfer them to air-gapped environments.

RKE2 does not come with Istio or Monitoring charts. If you're referring to the Rancher charts, this repo is not the correct place to ask about that. I will note that Rancher does not (in general) support direct installation of their charts - they are designed to be deployed via Rancher, with configuration specific to the downstream cluster injected into the chart values. But again, this is Rancher stuff, not RKE2.

**Comment by busyboy77** [comment]
Ideally, the ownership of RKE2 is under Rancher-- right?
even if this not the case, we should have a mechanism to download and install RKE2 Helm charts independently for the sake of air-gapped envs.

Consider delivering a solution with half air-gap possible ( RKE2 deployment etc )  and then you have to enable internet for 2nd half of the product line ( enable internet, install Rancher Server followed by installation of all the Apps menu items )

**Reply by brandond** [reply]
> the ownership of RKE2 is under Rancher

No

Both Rancher and RKE2 are SUSE products. Rancher is a different product, with a different repo, and engineers that work on RKE2 do not generally work on Rancher. You should take your inquiry to the correct repo so that it can be seen by the correct team.

Rancher has its own docs and process for managing airgap installs, and as I mentioned earlier, the Rancher charts are not meant for standalone use.

**Comment by busyboy77** [comment]
Got it -- let me open it with rancher folks
